File: Neptune(MC-ICP-MS)/Neptune_Data_Analyzer.py
#!/usr/bin/env python
# coding: utf-8

# In[1]:


# データを総合して解析を行う
import os
from Neptune_Function_d56Fe import read, Ratio, Delta #呼び出す関数は.py形式で保存しておく。
import pandas as pd
import numpy as np
get_ipython().run_line_magic('matplotlib', 'inline')
import matplotlib.pyplot as plt
os.chdir("自分のフォルダのパス")
print(os.getcwd())


# In[2]:


#blk + standard1
blk1_name=input('blk file.exp:')
std1_name=input('standard file.exp:')

blk1,std1 = read(blk1_name,std1_name)


# In[3]:


#blk + sample
blk2_name=input('blk file.exp:')
sample_name=input('sample file.exp:')

blk2,sample = read(blk2_name,sample_name)


# In[4]:


#blk + standard2
blk3_name=input('blk file.exp:')
std2_name=input('standard file.exp:')

blk3,std2 = read(blk3_name,std2_name)


# In[5]:


IRMM1 = Ratio(blk1,std1)
Sample = Ratio(blk2,sample)
IRMM2 = Ratio(blk3,std2)


# In[9]:


delta_values = Delta(IRMM1,Sample,IRMM2)
delta_values


# In[ ]:

File: Neptune(MC-ICP-MS)/Neptune_Data_Analyzer.R
#Neptune_functionの読み込み
.myfunc.env = new.env()
sys.source('~/自分のフォルダのパス/Neptune_function(d56Fe).R', envir=.myfunc.env)
attach(.myfunc.env)

#フォルダ選択
setwd("生データのフォルダのパス")

#生データの入力（基本的にはskipは固定でOK）
x1 <- read.csv("01-blk.exp",skip=16,nrows=20,sep="",row.names=1) # blk file
x2 <- read.csv("02-IRMM.exp",skip=16,nrows=60,sep="",row.names=1) # IRMM file
x3 <- read.csv("03-blk.exp",skip=16,nrows=20,sep="",row.names=1) # blk file
x4 <- read.csv("04-sample.exp",skip=16,nrows=60,sep="",row.names=1) # sample file
x5 <- read.csv("05-blk.exp",skip=16,nrows=20,sep="",row.names=1) # blk file
x6 <- read.csv("06-IRMM.exp",skip=16,nrows=60,sep="",row.names=1) # IRMM file

#計算
IRMM1  <- Ratio(x1,x2)
Sample <- Ratio(x3,x4)
IRMM2  <- Ratio(x5,x6)
delta_values <- Delta(IRMM1,Sample,IRMM2)
print(delta_values)

#結果の保存
write.table(delta_values,"result.csv",append=T,col.names=NA,row.names=T,sep=",",quote=F)
